scripts/background.js

```diff
@@ -291,29 +372,28 @@ function beforeRequest(details) {
                 storeNewTokens(tokens, signedPoints);
             }
             // Reload the page for the originally intended url
-            let url = new URL(reqUrl);
-            if (url.href.indexOf(CHL_CAPTCHA_DOMAIN) == -1){
-                let captchaPath = url.pathname;
-                let pathIndex = url.href.indexOf(captchaPath);
-                let reloadUrl = url.href.substring(0, pathIndex+1);
-                setSpendFlag(reloadUrl, true);
-                chrome.tabs.update(details.tabId, { url: reloadUrl });
+            if (RELOAD_ON_SIGN) {
+                let url = new URL(reqUrl);
+                if (url.href.indexOf(CHL_CAPTCHA_DOMAIN) == -1){
+                    let captchaPath = url.pathname;
+                    let pathIndex = url.href.indexOf(captchaPath);
+                    let reloadUrl = url.href.substring(0, pathIndex+1);
+                    setSpendFlag(url.host, true);
+                    chrome.tabs.update(tabId, { url: reloadUrl });
+                }
             }
         } else if (countStoredTokens() >= (MAX_TOKENS - TOKENS_PER_REQUEST)) {
-            throw new Error("[privacy-pass]: Cannot receive new tokens due to upper bound.")
+            throw new Error("[privacy-pass]: Cannot receive new tokens due to upper bound.");
         }
     };
 
     xhr.open("POST", newUrl, true);
     xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
-    xhr.setRequestHeader("CF-Chl-Bypass", "1");
+    xhr.setRequestHeader(CHL_BYPASS_SUPPORT, "1");
     // We seem to get back some odd mime types that cause problems...
     xhr.overrideMimeType("text/plain");
 
-    xhr.send("blinded-tokens=" + request);
-
-    // Cancel the original request
-    return {redirectUrl: "javascript:void(0)"};
+    xhr.send(requestBody);
```

Comment:
Why did we decide not to cancel the original request after sending the blinded tokens requestBody?
The code removed was 
```
    // Cancel the original request	
    return {redirectUrl: "javascript:void(0)"};
```

Reply:
Good point. I replaced it with `return {cancel: true}` in the calling function, but the reason that we did it like this is to prevent a holding page appearing in Chrome. I've reinstated it now.